2024-06-18 09:42:42,147: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2024-06-18 09:42:42,156: alpaca.trading.stream: INFO: connected to: BaseURL.TRADING_STREAM_PAPER
2024-06-18 09:42:43,489: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-06-18 09:42:43,489: asyncio: INFO: [MLTrader] Executing the before_market_opens lifecycle method
2024-06-18 09:42:43,490: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-06-18 09:42:43,490: asyncio: INFO: [MLTrader] Executing the before_starting_trading lifecycle method
2024-06-18 09:42:43,492: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2024-06-18 09:42:43[0m
2024-06-18 09:42:43,492: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-06-18 09:42:44,322: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-06-18 09:42:44,323: asyncio: ERROR: [MLTrader] 'datetime.datetime' object has no attribute 'strfttime'
2024-06-18 09:42:44,326: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 923, in run
    self._run_trading_session()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 871, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 286, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 397, in _on_trading_iteration
    raise e
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 369, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\.vscode\pythonCode\tradingBot.py", line 52, in on_trading_iteration
    news = self.get_news()
           ^^^^^^^^^^^^^^^
  File "C:\Users\cordo\.vscode\pythonCode\tradingBot.py", line 40, in get_news
    today, three_days_prior = self.get_dates()
                              ^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\.vscode\pythonCode\tradingBot.py", line 37, in get_dates
    return today.strfttime('%Y-%m-%d'), three_days_prior.strfttime('%Y-%m-%d')
           ^^^^^^^^^^^^^^^
AttributeError: 'datetime.datetime' object has no attribute 'strfttime'. Did you mean: 'strftime'?

2024-06-18 09:42:44,329: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2024-06-18 09:42:44,341: root: INFO: Backtesting finished
2024-06-18 09:42:44,654: asyncio: WARNING: [MLTrader] Cannot plot returns because the strategy returns are missing
2024-06-18 09:42:44,796: asyncio: WARNING: [MLTrader] Cannot create a tearsheet because the strategy returns are missing
2024-06-18 09:42:44,796: backtest_stats: INFO: Backtest took 0:00:02.651545 for a speed of 0.000
